src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/response/PaperResponseReviewedCallbackHandler.java

```diff
@@ -0,0 +1,257 @@
+        ClaimDocumentType.PAPER_RESPONSE_DISPUTES_ALL,
+        ClaimDocumentType.PAPER_RESPONSE_PART_ADMIT,
+        ClaimDocumentType.PAPER_RESPONSE_STATES_PAID);
+
+    private static List<String> paperResponseScannedType = Arrays.asList("N9a",
+        "N9b",
+        "N11",
+        "N225",
+        "N180");
+
+    private static Function<Claim, List<ClaimDocument>> getStaffUploadedDocs = claim ->
+        claim.getClaimDocumentCollection()
+        .map(ClaimDocumentCollection::getStaffUploadedDocuments)
+        .orElse(Collections.emptyList());
+
+    private static Function<Claim, List<ScannedDocument>> getBulkScannedDocs = claim ->
+        claim.getClaimDocumentCollection()
+        .map(ClaimDocumentCollection::getScannedDocuments)
+        .orElse(Collections.emptyList());
+
+    private static Predicate<ClaimDocument> filterClaimDocumentPaperResponseDoc = doc ->
+        paperResponseStaffUploadedType.stream()
+            .anyMatch(type -> type.equals(doc.getDocumentType()));
+
+    private static Predicate<ScannedDocument> filterScannedDocumentPaperResponseDoc = doc ->
+        paperResponseScannedType.stream().anyMatch(type -> type.equalsIgnoreCase(doc.getSubtype()));
+
+    private static Predicate<ClaimDocument> isClaimDocumentMoreTimeRequested = doc ->
+        doc.getDocumentType().equals(ClaimDocumentType.PAPER_RESPONSE_MORE_TIME);
+
+    private static Predicate<ScannedDocument> isScannedDocumentMoreTimeRequested = doc ->
+        doc.getSubtype().equalsIgnoreCase("N9");
+
+    private final ResponseDeadlineCalculator responseDeadlineCalculator;
+    private final MoreTimeRequestRule moreTimeRequestRule;
+
+    @Autowired
+    public PaperResponseReviewedCallbackHandler(
+        CaseDetailsConverter caseDetailsConverter,
+        CaseMapper caseMapper,
+        ResponseDeadlineCalculator responseDeadlineCalculator,
+        MoreTimeRequestRule moreTimeRequestRule) {
+        this.caseDetailsConverter = caseDetailsConverter;
+        this.caseMapper = caseMapper;
+        this.responseDeadlineCalculator = responseDeadlineCalculator;
+        this.moreTimeRequestRule = moreTimeRequestRule;
+    }
+
+    @Override
+    protected Map<CallbackType, Callback> callbacks() {
+        return ImmutableMap.of(
+            CallbackType.ABOUT_TO_START, this::verifyResponsePossible,
+            CallbackType.ABOUT_TO_SUBMIT, this::updateResponseDeadline
+        );
+    }
+
+    @Override
+    public List<CaseEvent> handledEvents() {
+        return Collections.singletonList(CaseEvent.REVIEWED_PAPER_RESPONSE);
+    }
+
+    @Override
+    public List<Role> getSupportedRoles() {
+        return ROLES;
+    }
+
+    private AboutToStartOrSubmitCallbackResponse updateResponseDeadline(CallbackParams callbackParams) {
+        CallbackRequest callbackRequest = callbackParams.getRequest();
+        Claim claim = toClaimAfterEvent(callbackRequest);
+
+        if (verifyNoDuplicateMoreTimeRequested(callbackRequest)) {
```

Comment:
as i was saying in DM just as suggestion, could we maybe extract `verifyNoDuplicateMoreTimeRequested` and all related code to other helper class!!